sycl/test/esimd/memory_properties.cpp

```diff
@@ -531,4 +531,31 @@ SYCL_ESIMD_FUNCTION SYCL_EXTERNAL void test_block_store(AccType &acc,
 
   // CHECK: call void @llvm.genx.oword.st.v4i32(i32 {{[^)]+}}, i32 {{[^)]+}}, <4 x i32> {{[^)]+}})
   block_store(acc, byte_offset32, valsi, store_props_b);
+
+  // Now try SLM block_store() with and without cache hints that are ignored.
+
+  // CHECK: store <4 x float> {{[^)]+}}, ptr addrspace(3) {{[^)]+}}, align 16
+  slm_block_store(byte_offset32, vals, store_props_b);
+
+  // CHECK: store <4 x float> {{[^)]+}}, ptr addrspace(3) {{[^)]+}}, align 16
+  slm_block_store(byte_offset32, vals, store_props_a);
+
+  // Now try SLM block_store() with a predicate.
+
+  // CHECK: call void @llvm.genx.lsc.store.slm.v1i1.v1i32.v4i32(<1 x i1> {{[^)]+}}, i8 4, i8 0, i8 0, i16 1, i32 0, i8 3, i8 4, i8 2, i8 0, <1 x i32> {{[^)]+}}, <4 x i32> {{[^)]+}}, i32 0)
+  slm_block_store(byte_offset32, valsi, mask, store_props_b);
+
+  // Now try block_store() accepting local accessor.
+
+  // CHECK: store <4 x float> {{[^)]+}}, ptr addrspace(3) {{[^)]+}}, align 8
+  block_store(local_acc, vals, store_props_d);
+
+  // CHECK: store <4 x i32> {{[^)]+}}, ptr addrspace(3) {{[^)]+}}, align 8
+  block_store(local_acc, byte_offset32, valsi, store_props_d);
+
+  // CHECK: call void @llvm.genx.lsc.store.slm.v1i1.v1i32.v4f32(<1 x i1> {{[^)]+}}, i8 4, i8 0, i8 0, i16 1, i32 0, i8 3, i8 4, i8 2, i8 0, <1 x i32> {{[^)]+}}, <4 x float> {{[^)]+}}, i32 0)
+  block_store(local_acc, vals, mask, store_props_a);
+
+  // CHECK: call void @llvm.genx.lsc.store.slm.v1i1.v1i32.v4i32(<1 x i1> {{[^)]+}}, i8 4, i8 0, i8 0, i16 1, i32 0, i8 3, i8 4, i8 2, i8 0, <1 x i32> {{[^)]+}}, <4 x i32> {{[^)]+}}, i32 0)
+  block_store(local_acc, byte_offset32, valsi, mask, store_props_c);
```

Comment:
E2E tests do not verify the case when `vals` is `simd_view`. Please add such checks here.